src/common/com/intellij/plugins/haxe/ide/HaxeCompilerCompletionContributor.java

```diff
@@ -120,16 +123,24 @@ protected void addCompletions(@NotNull CompletionParameters parameters,
                        //Export/flash/haxe contains build.hxml which gets generated after build
                        break;
                      case HaxeModuleSettingsBaseImpl.USE_OPENFL:
-                       commandLineArguments.add(HaxelibCommandUtils.getHaxelibPath(moduleForFile));
-                       commandLineArguments.add("run");
-                       commandLineArguments.add("lime");
-                       commandLineArguments.add("display");
-                       //flash, html5, linux, etc
-                       String targetFlag = moduleSettings.getNmeTarget().getTargetFlag();
-                       commandLineArguments.add(targetFlag);
-
-                       List<String> stdout = HaxelibCommandUtils.getProcessStdout(commandLineArguments,
-                                                                                  BuildProperties.getProjectBaseDir(project));
+                       String targetFlag = moduleSettings.getOpenFLTarget().getTargetFlag();
+
+                       List<String> stdout = openFLDisplayArguments.get(moduleForFile.getModuleFilePath() + targetFlag);
+                       if (stdout == null) {
+                         commandLineArguments.add(HaxelibCommandUtils.getHaxelibPath(moduleForFile));
+                         commandLineArguments.add("run");
+                         commandLineArguments.add("lime");
+                         commandLineArguments.add("display");
+                         //flash, html5, linux, etc
+
+                         commandLineArguments.add(targetFlag);
+
+                         stdout = HaxelibCommandUtils.getProcessStdout(commandLineArguments,
+                                                                       BuildProperties.getProjectBaseDir(project));
+
+                         openFLDisplayArguments.put(moduleForFile.getModuleFilePath() + targetFlag, stdout);
```

Comment:
I think I made it not for currently edited file, but for module, because targets can vary between modules